### PR TITLE
NMS-15759: Allow metadata for notification credentials

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/NMS-15759

### CODE REVIEW

File: docs/modules/reference/pages/daemons/daemon-config-files/notifd.adoc [hunk 1]
```diff
@@ -44,7 +44,17 @@ Generates xref:operation:deep-dive/notifications/introduction.adoc[notifications
 | no
 
 | javamail-configuration.properties
-| Configuration to send notification mails via specific mail servers.
+| Configuration to send notification mails via specific mail servers. Metadata expression can be used for the authenticateUser and authenticatePassword fields in order to reference credentials stored in the secure credentials vault.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/daemons/daemon-config-files/notifd.adoc:47]:
```suggestion
| Configuration to send notification emails via specific mail servers. Metadata expression can be used for the authenticateUser and authenticatePassword fields in order to reference credentials stored in the secure credentials vault.
```

File: docs/modules/reference/pages/daemons/daemon-config-files/notifd.adoc [hunk 2]
```diff
@@ -44,7 +44,17 @@ Generates xref:operation:deep-dive/notifications/introduction.adoc[notifications
...
+| no
+| no
+
+| microblog-configuration.properties
+| Configuration to send notification mails via specific mail servers. Metadata expression can be used for the authen-* and oauth-* fields in order to reference credentials stored in the secure credentials vault.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/daemons/daemon-config-files/notifd.adoc:52]:
```suggestion
| Configuration to send notification emails via specific mail servers. Metadata expression can be used for the authen-* and oauth-* fields in order to reference credentials stored in the secure credentials vault.
```

File: docs/modules/reference/pages/daemons/daemon-config-files/notifd.adoc [hunk 3]
```diff
@@ -44,7 +44,17 @@ Generates xref:operation:deep-dive/notifications/introduction.adoc[notifications
...
+| no
+| no
+
+| xmpp-configuration.properties
+| Configuration to send notification mails via specific mail servers. Metadata expression can be used for the xmpp.user and xmpp.pass fields in order to reference credentials stored in the secure credentials vault.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/daemons/daemon-config-files/notifd.adoc:57]:
```suggestion
| Configuration to send notification emails via specific mail servers. Metadata expression can be used for the xmpp.user and xmpp.pass fields in order to reference credentials stored in the secure credentials vault.
```

File: docs/modules/reference/pages/daemons/daemon-config-files/notifd.adoc (hunk 1 above)
Comment (RangerRick) [reply docs/modules/reference/pages/daemons/daemon-config-files/notifd.adoc:47]: Also should have carriage-returns between sentences, right? That's one style thing I _do_ know. :D

**Pull request merged**